forge_sync/resources.py

```diff
@@ -2,19 +2,22 @@
 from typing import Dict
 
 import pymongo
-from dagster import ConfigurableResource, Definitions
+from dagster import ConfigurableIOManager
 
 
-mongo_client = pymongo.MongoClient(os.getenv("MONGO_URI"))
-
-
-class MongoDBResource(ConfigurableResource):
+class MongoIO(ConfigurableIOManager):
```

Comment:
Alguma razão de preferência entre IO vs Configurable Resource? De acordo com a docs: https://docs.dagster.io/concepts/io-management/io-managers#io-managers

IO managers seriam interfaces entre assets ou ops:
![image](https://github.com/DataForgeFin/forge-sync/assets/20599467/3e92fa53-4a89-40ae-bba5-22b8fb14ad36)

O que me leva a crer que é uma forma de passar informação entre "tasks" e não uma forma reutilizável de gerenciamento de data sinks como visto aqui: https://docs.dagster.io/concepts/resources#resources.

Outro ponto: a adição da propriedade intrínseca da classe `connection_string` acaba deixando a senha do banco em plain text string na UI: http://127.0.0.1:3000/locations/forge_sync/resources/io_manager algo que achei ruim pois da forma anterior, um usuário da aplicação do dagster não teria acesso a senha de configuração do banco (se houvesse uma forma de deixar isso escondido seria melhor).

Por fim, testei essa PR e esse IO Manager não está conseguindo conectar ao banco e inserir o dado com o seguinte erro:

```
2023-06-24 11:32:47 -0300 - dagster - DEBUG - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - 67726 - RUN_START - Started execution of run for "__ASSET_JOB".
2023-06-24 11:32:47 -0300 - dagster - DEBUG - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - 67726 - ENGINE_EVENT - Executing steps using multiprocess executor: parent process (pid: 67726)
2023-06-24 11:32:47 -0300 - dagster - DEBUG - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - 67726 - selic - STEP_WORKER_STARTING - Launching subprocess for "selic".
2023-06-24 11:32:49 -0300 - dagster - DEBUG - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - 67736 - STEP_WORKER_STARTED - Executing step "selic" in subprocess.
2023-06-24 11:32:49 -0300 - dagster - DEBUG - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - 67736 - selic - RESOURCE_INIT_STARTED - Starting initialization of resources [io_manager].
2023-06-24 11:32:49 -0300 - dagster - DEBUG - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - 67736 - selic - RESOURCE_INIT_SUCCESS - Finished initialization of resources [io_manager].
2023-06-24 11:32:49 -0300 - dagster - DEBUG - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - 67736 - LOGS_CAPTURED - Started capturing logs in process (pid: 67736).
2023-06-24 11:32:49 -0300 - dagster - DEBUG - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - 67736 - selic - STEP_START - Started execution of step "selic".
2023-06-24 11:32:49 -0300 - dagster - INFO - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - selic - Found {'by': 'current', 'valid_key': True, 'results': [{'date': '2023-06-22', 'cdi': 13.75, 'selic': 13.75, 'daily_factor': 1.00050788, 'selic_daily': 13.65, 'cdi_daily': 13.65}], 'execution_time': 0.0, 'from_cache': True}
2023-06-24 11:32:49 -0300 - dagster - DEBUG - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - 67736 - selic - STEP_OUTPUT - Yielded output "result" of type "Any". (Type check passed).
2023-06-24 11:32:49 -0300 - dagster - INFO - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - selic - Saving results
2023-06-24 11:32:57 -0300 - dagster.daemon.SensorDaemon - INFO - Not checking for any runs since no sensors have been started.
2023-06-24 11:33:19 -0300 - dagster - ERROR - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - 67736 - selic - STEP_FAILURE - Execution of step "selic" failed.

dagster._core.errors.DagsterExecutionHandleOutputError: Error occurred while handling output "result" of step "selic"::

pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [Errno 61] Connection refused, Timeout: 30s, Topology Description: <TopologyDescription id: 6496fe9105ae455adaef107a, topology_type: Unknown, servers: [<ServerDescription ('localhost', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('localhost:27017: [Errno 61] Connection refused')>]>

Stack Trace:
  File "/Users/andre/projects/forge-sync/.venv/lib/python3.8/site-packages/dagster/_core/execution/plan/utils.py", line 54, in op_execution_error_boundary
    yield
  File "/Users/andre/projects/forge-sync/.venv/lib/python3.8/site-packages/dagster/_utils/__init__.py", line 445, in iterate_with_context
    next_output = next(iterator)
  File "/Users/andre/projects/forge-sync/.venv/lib/python3.8/site-packages/dagster/_core/execution/plan/execute_step.py", line 593, in _gen_fn
    gen_output = output_manager.handle_output(output_context, output.value)
  File "/Users/andre/projects/forge-sync/forge_sync/resources.py", line 18, in handle_output
    collection.insert_one(obj)
  File "/Users/andre/projects/forge-sync/.venv/lib/python3.8/site-packages/pymongo/collection.py", line 628, in insert_one
    self._insert_one(
  File "/Users/andre/projects/forge-sync/.venv/lib/python3.8/site-packages/pymongo/collection.py", line 569, in _insert_one
    self.__database.client._retryable_write(acknowledged, _insert_command, session)
  File "/Users/andre/projects/forge-sync/.venv/lib/python3.8/site-packages/pymongo/mongo_client.py", line 1475, in _retryable_write
    with self._tmp_session(session) as s:
  File "/Users/andre/.pyenv/versions/3.8.13/lib/python3.8/contextlib.py", line 113, in __enter__
    return n2023-06-24 11:33:20 -0300 - dagster - DEBUG - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - 67726 - ENGINE_EVENT - Multiprocess executor: parent process exiting after 32.71s (pid: 67726)
2023-06-24 11:33:20 -0300 - dagster - ERROR - __ASSET_JOB - 5609b0d8-d8cc-43f6-87b5-b354a548d564 - 67726 - RUN_FAILURE - Execution of run for "__ASSET_JOB" failed. Steps failed: ['selic'].
```
Me parece que a connection string esteja errada no IO? Apesar de ela estar correta na UI com a senha e user e host (PS: testei a implementação da PR anterior entre essa execução e a conexão foi bem sucedida o que me indica que é um erro de config no código).

Na UI mostra que o IO manager não é usado por ninguém:
![image](https://github.com/DataForgeFin/forge-sync/assets/20599467/724b47c9-b809-45e2-976a-18e9931e000c)

Reply:
IO Manager é um tipo de `Resource`, porém com uma interface especializada para entrada e saída das `Ops`. Ao adotarmos IO Manager vamos conseguir isolar a lógica de geração do `Asset` da sua materialização. Se quisermos otimizar a escrita no MongoDB ou trocar por outro banco, faremos isso alterando apenas o IO Manager, sem alterar o código do `Asset`.

Reply:
Vou remover a propriedade da connection string para deixar de expor na UI.

Reply:
O erro que você recebeu era devido um bug corrigido em 1521c82612e6ff2fee1f9c75adbd8bb980c495b6

Reply:
> Vou remover a propriedade da connection string para deixar de expor na UI.

Inclusive esse é um bom caso de uso para `Resource`.